	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.client.MetaScanner.metaScan(MetaScanner.java:144)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.prefetchRegionCache(HConnectionManager.java:1140)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegionInMeta(HConnectionManager.java:1202)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1092)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1049)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.getRegionLocation(HConnectionManager.java:890)
	at org.apache.hadoop.hbase.client.RegionServerCallable.prepare(RegionServerCallable.java:72)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:116)
	at org.apache.hadoop.hbase.client.HTable.get(HTable.java:787)
	at org.apache.hadoop.hbase.examples.customer.GetTable.main(GetTable.java:33)
[17:17:17,472] [main] DEBUG (RpcClient.java:1017) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closed
[17:17:17,472] [main]  INFO (RpcRetryingCaller.java:120) - Call exception, tries=22, retries=35, retryTime=309517ms, msg=row 'Customer,101,99999999999999' on table 'hbase:meta' at region=hbase:meta,,1.1588230740, hostname=localhost,40228,1417408326094, seqNum=0
[17:17:30,806] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:717) - Got ping response for sessionid: 0x14a041ea3cd0030 after 1ms
[17:17:37,495] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 49,4  replyHeader:: 49,260,0  request:: '/hbase/table/hbase:meta,F  response:: #ffffffff000146d61737465723a3436373430ffffffbdffffffa98ffffff83fffffff8ffffffe83fffffffe65042554680,s{36,37,1417408333505,1417408333517,1,0,0,0,31,0,36} 
[17:17:37,499] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 50,4  replyHeader:: 50,260,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a343032323829fffffff153cffffffbf6ffffffd8ffffffc250425546a16a96c6f63616c686f737410ffffffa4ffffffba218ffffffceffffffd3fffffffaffffffa0ffffffa029100,s{39,39,1417408333631,1417408333631,0,0,0,0,61,0,39} 
[17:17:37,500] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[17:17:37,501] [main] DEBUG (RpcClient.java:862) - Connecting to localhost/127.0.0.1:40228
[17:17:37,503] [main] DEBUG (RpcClient.java:1009) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closing ipc connection to localhost/127.0.0.1:40228: Connection refused
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:206)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:529)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:493)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupConnection(RpcClient.java:578)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:868)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.client.MetaScanner.metaScan(MetaScanner.java:144)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.prefetchRegionCache(HConnectionManager.java:1140)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegionInMeta(HConnectionManager.java:1202)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1092)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1049)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.getRegionLocation(HConnectionManager.java:890)
	at org.apache.hadoop.hbase.client.RegionServerCallable.prepare(RegionServerCallable.java:72)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:116)
	at org.apache.hadoop.hbase.client.HTable.get(HTable.java:787)
	at org.apache.hadoop.hbase.examples.customer.GetTable.main(GetTable.java:33)
[17:17:37,505] [main] DEBUG (RpcClient.java:1017) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closed
[17:17:37,506] [main]  INFO (RpcRetryingCaller.java:120) - Call exception, tries=23, retries=35, retryTime=329551ms, msg=row 'Customer,101,99999999999999' on table 'hbase:meta' at region=hbase:meta,,1.1588230740, hostname=localhost,40228,1417408326094, seqNum=0
[17:17:50,844] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:717) - Got ping response for sessionid: 0x14a041ea3cd0030 after 1ms
[17:17:57,631] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 51,4  replyHeader:: 51,260,0  request:: '/hbase/table/hbase:meta,F  response:: #ffffffff000146d61737465723a3436373430ffffffbdffffffa98ffffff83fffffff8ffffffe83fffffffe65042554680,s{36,37,1417408333505,1417408333517,1,0,0,0,31,0,36} 
[17:17:57,632] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 52,4  replyHeader:: 52,260,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a343032323829fffffff153cffffffbf6ffffffd8ffffffc250425546a16a96c6f63616c686f737410ffffffa4ffffffba218ffffffceffffffd3fffffffaffffffa0ffffffa029100,s{39,39,1417408333631,1417408333631,0,0,0,0,61,0,39} 
[17:17:57,632] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[17:17:57,633] [main] DEBUG (RpcClient.java:862) - Connecting to localhost/127.0.0.1:40228
[17:17:57,633] [main] DEBUG (RpcClient.java:1009) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closing ipc connection to localhost/127.0.0.1:40228: Connection refused
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:206)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:529)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:493)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupConnection(RpcClient.java:578)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:868)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.client.MetaScanner.metaScan(MetaScanner.java:144)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.prefetchRegionCache(HConnectionManager.java:1140)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegionInMeta(HConnectionManager.java:1202)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1092)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1049)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.getRegionLocation(HConnectionManager.java:890)
	at org.apache.hadoop.hbase.client.RegionServerCallable.prepare(RegionServerCallable.java:72)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:116)
	at org.apache.hadoop.hbase.client.HTable.get(HTable.java:787)
	at org.apache.hadoop.hbase.examples.customer.GetTable.main(GetTable.java:33)
[17:17:57,635] [main] DEBUG (RpcClient.java:1017) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closed
[17:17:57,635] [main]  INFO (RpcRetryingCaller.java:120) - Call exception, tries=24, retries=35, retryTime=349680ms, msg=row 'Customer,101,99999999999999' on table 'hbase:meta' at region=hbase:meta,,1.1588230740, hostname=localhost,40228,1417408326094, seqNum=0
[17:18:10,973] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:717) - Got ping response for sessionid: 0x14a041ea3cd0030 after 0ms
[17:18:17,832] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 53,4  replyHeader:: 53,260,0  request:: '/hbase/table/hbase:meta,F  response:: #ffffffff000146d61737465723a3436373430ffffffbdffffffa98ffffff83fffffff8ffffffe83fffffffe65042554680,s{36,37,1417408333505,1417408333517,1,0,0,0,31,0,36} 
[17:18:17,835] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 54,4  replyHeader:: 54,260,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a343032323829fffffff153cffffffbf6ffffffd8ffffffc250425546a16a96c6f63616c686f737410ffffffa4ffffffba218ffffffceffffffd3fffffffaffffffa0ffffffa029100,s{39,39,1417408333631,1417408333631,0,0,0,0,61,0,39} 
[17:18:17,836] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[17:18:17,837] [main] DEBUG (RpcClient.java:862) - Connecting to localhost/127.0.0.1:40228
[17:18:17,839] [main] DEBUG (RpcClient.java:1009) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closing ipc connection to localhost/127.0.0.1:40228: Connection refused
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:206)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:529)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:493)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupConnection(RpcClient.java:578)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:868)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.client.MetaScanner.metaScan(MetaScanner.java:144)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.prefetchRegionCache(HConnectionManager.java:1140)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegionInMeta(HConnectionManager.java:1202)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1092)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1049)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.getRegionLocation(HConnectionManager.java:890)
	at org.apache.hadoop.hbase.client.RegionServerCallable.prepare(RegionServerCallable.java:72)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:116)
	at org.apache.hadoop.hbase.client.HTable.get(HTable.java:787)
	at org.apache.hadoop.hbase.examples.customer.GetTable.main(GetTable.java:33)
[17:18:17,840] [main] DEBUG (RpcClient.java:1017) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closed
[17:18:17,841] [main]  INFO (RpcRetryingCaller.java:120) - Call exception, tries=25, retries=35, retryTime=369886ms, msg=row 'Customer,101,99999999999999' on table 'hbase:meta' at region=hbase:meta,,1.1588230740, hostname=localhost,40228,1417408326094, seqNum=0
[17:18:31,175] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:717) - Got ping response for sessionid: 0x14a041ea3cd0030 after 1ms
[17:18:37,983] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 55,4  replyHeader:: 55,260,0  request:: '/hbase/table/hbase:meta,F  response:: #ffffffff000146d61737465723a3436373430ffffffbdffffffa98ffffff83fffffff8ffffffe83fffffffe65042554680,s{36,37,1417408333505,1417408333517,1,0,0,0,31,0,36} 
[17:18:37,986] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 56,4  replyHeader:: 56,260,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a343032323829fffffff153cffffffbf6ffffffd8ffffffc250425546a16a96c6f63616c686f737410ffffffa4ffffffba218ffffffceffffffd3fffffffaffffffa0ffffffa029100,s{39,39,1417408333631,1417408333631,0,0,0,0,61,0,39} 
[17:18:37,987] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[17:18:37,988] [main] DEBUG (RpcClient.java:862) - Connecting to localhost/127.0.0.1:40228
[17:18:37,989] [main] DEBUG (RpcClient.java:1009) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closing ipc connection to localhost/127.0.0.1:40228: Connection refused
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:206)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:529)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:493)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupConnection(RpcClient.java:578)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:868)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.client.MetaScanner.metaScan(MetaScanner.java:144)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.prefetchRegionCache(HConnectionManager.java:1140)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegionInMeta(HConnectionManager.java:1202)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1092)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1049)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.getRegionLocation(HConnectionManager.java:890)
	at org.apache.hadoop.hbase.client.RegionServerCallable.prepare(RegionServerCallable.java:72)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:116)
	at org.apache.hadoop.hbase.client.HTable.get(HTable.java:787)
	at org.apache.hadoop.hbase.examples.customer.GetTable.main(GetTable.java:33)
[17:18:37,991] [main] DEBUG (RpcClient.java:1017) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closed
[17:18:37,992] [main]  INFO (RpcRetryingCaller.java:120) - Call exception, tries=26, retries=35, retryTime=390036ms, msg=row 'Customer,101,99999999999999' on table 'hbase:meta' at region=hbase:meta,,1.1588230740, hostname=localhost,40228,1417408326094, seqNum=0
[17:18:51,334] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:717) - Got ping response for sessionid: 0x14a041ea3cd0030 after 1ms
[17:18:58,054] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 57,4  replyHeader:: 57,260,0  request:: '/hbase/table/hbase:meta,F  response:: #ffffffff000146d61737465723a3436373430ffffffbdffffffa98ffffff83fffffff8ffffffe83fffffffe65042554680,s{36,37,1417408333505,1417408333517,1,0,0,0,31,0,36} 
[17:18:58,057] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 58,4  replyHeader:: 58,260,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a343032323829fffffff153cffffffbf6ffffffd8ffffffc250425546a16a96c6f63616c686f737410ffffffa4ffffffba218ffffffceffffffd3fffffffaffffffa0ffffffa029100,s{39,39,1417408333631,1417408333631,0,0,0,0,61,0,39} 
[17:18:58,058] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[17:18:58,059] [main] DEBUG (RpcClient.java:862) - Connecting to localhost/127.0.0.1:40228
[17:18:58,060] [main] DEBUG (RpcClient.java:1009) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closing ipc connection to localhost/127.0.0.1:40228: Connection refused
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:206)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:529)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:493)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupConnection(RpcClient.java:578)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:868)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.client.MetaScanner.metaScan(MetaScanner.java:144)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.prefetchRegionCache(HConnectionManager.java:1140)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegionInMeta(HConnectionManager.java:1202)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1092)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1049)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.getRegionLocation(HConnectionManager.java:890)
	at org.apache.hadoop.hbase.client.RegionServerCallable.prepare(RegionServerCallable.java:72)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:116)
	at org.apache.hadoop.hbase.client.HTable.get(HTable.java:787)
	at org.apache.hadoop.hbase.examples.customer.GetTable.main(GetTable.java:33)
[17:18:58,062] [main] DEBUG (RpcClient.java:1017) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closed
[17:18:58,063] [main]  INFO (RpcRetryingCaller.java:120) - Call exception, tries=27, retries=35, retryTime=410107ms, msg=row 'Customer,101,99999999999999' on table 'hbase:meta' at region=hbase:meta,,1.1588230740, hostname=localhost,40228,1417408326094, seqNum=0
[17:19:11,400] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:717) - Got ping response for sessionid: 0x14a041ea3cd0030 after 1ms
[17:19:18,231] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 59,4  replyHeader:: 59,260,0  request:: '/hbase/table/hbase:meta,F  response:: #ffffffff000146d61737465723a3436373430ffffffbdffffffa98ffffff83fffffff8ffffffe83fffffffe65042554680,s{36,37,1417408333505,1417408333517,1,0,0,0,31,0,36} 
[17:19:18,235] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 60,4  replyHeader:: 60,260,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a343032323829fffffff153cffffffbf6ffffffd8ffffffc250425546a16a96c6f63616c686f737410ffffffa4ffffffba218ffffffceffffffd3fffffffaffffffa0ffffffa029100,s{39,39,1417408333631,1417408333631,0,0,0,0,61,0,39} 
[17:19:18,236] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[17:19:18,237] [main] DEBUG (RpcClient.java:862) - Connecting to localhost/127.0.0.1:40228
[17:19:18,238] [main] DEBUG (RpcClient.java:1009) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closing ipc connection to localhost/127.0.0.1:40228: Connection refused
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:206)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:529)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:493)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupConnection(RpcClient.java:578)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:868)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.client.MetaScanner.metaScan(MetaScanner.java:144)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.prefetchRegionCache(HConnectionManager.java:1140)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegionInMeta(HConnectionManager.java:1202)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1092)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1049)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.getRegionLocation(HConnectionManager.java:890)
	at org.apache.hadoop.hbase.client.RegionServerCallable.prepare(RegionServerCallable.java:72)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:116)
	at org.apache.hadoop.hbase.client.HTable.get(HTable.java:787)
	at org.apache.hadoop.hbase.examples.customer.GetTable.main(GetTable.java:33)
[17:19:18,240] [main] DEBUG (RpcClient.java:1017) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closed
[17:19:18,240] [main]  INFO (RpcRetryingCaller.java:120) - Call exception, tries=28, retries=35, retryTime=430285ms, msg=row 'Customer,101,99999999999999' on table 'hbase:meta' at region=hbase:meta,,1.1588230740, hostname=localhost,40228,1417408326094, seqNum=0
[17:19:31,582] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:717) - Got ping response for sessionid: 0x14a041ea3cd0030 after 1ms
[17:19:38,318] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 61,4  replyHeader:: 61,262,0  request:: '/hbase/table/hbase:meta,F  response:: #ffffffff000146d61737465723a3436373430ffffffbdffffffa98ffffff83fffffff8ffffffe83fffffffe65042554680,s{36,37,1417408333505,1417408333517,1,0,0,0,31,0,36} 
[17:19:38,322] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 62,4  replyHeader:: 62,262,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a343032323829fffffff153cffffffbf6ffffffd8ffffffc250425546a16a96c6f63616c686f737410ffffffa4ffffffba218ffffffceffffffd3fffffffaffffffa0ffffffa029100,s{39,39,1417408333631,1417408333631,0,0,0,0,61,0,39} 
[17:19:38,323] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[17:19:38,324] [main] DEBUG (RpcClient.java:862) - Connecting to localhost/127.0.0.1:40228
[17:19:38,325] [main] DEBUG (RpcClient.java:1009) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closing ipc connection to localhost/127.0.0.1:40228: Connection refused
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:206)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:529)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:493)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupConnection(RpcClient.java:578)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:868)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.client.MetaScanner.metaScan(MetaScanner.java:144)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.prefetchRegionCache(HConnectionManager.java:1140)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegionInMeta(HConnectionManager.java:1202)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1092)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1049)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.getRegionLocation(HConnectionManager.java:890)
	at org.apache.hadoop.hbase.client.RegionServerCallable.prepare(RegionServerCallable.java:72)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:116)
	at org.apache.hadoop.hbase.client.HTable.get(HTable.java:787)
	at org.apache.hadoop.hbase.examples.customer.GetTable.main(GetTable.java:33)
[17:19:38,327] [main] DEBUG (RpcClient.java:1017) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closed
[17:19:38,327] [main]  INFO (RpcRetryingCaller.java:120) - Call exception, tries=29, retries=35, retryTime=450372ms, msg=row 'Customer,101,99999999999999' on table 'hbase:meta' at region=hbase:meta,,1.1588230740, hostname=localhost,40228,1417408326094, seqNum=0
[17:19:51,658] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:717) - Got ping response for sessionid: 0x14a041ea3cd0030 after 1ms
[17:19:58,349] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 63,4  replyHeader:: 63,262,0  request:: '/hbase/table/hbase:meta,F  response:: #ffffffff000146d61737465723a3436373430ffffffbdffffffa98ffffff83fffffff8ffffffe83fffffffe65042554680,s{36,37,1417408333505,1417408333517,1,0,0,0,31,0,36} 
[17:19:58,353] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 64,4  replyHeader:: 64,262,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a343032323829fffffff153cffffffbf6ffffffd8ffffffc250425546a16a96c6f63616c686f737410ffffffa4ffffffba218ffffffceffffffd3fffffffaffffffa0ffffffa029100,s{39,39,1417408333631,1417408333631,0,0,0,0,61,0,39} 
[17:19:58,354] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[17:19:58,355] [main] DEBUG (RpcClient.java:862) - Connecting to localhost/127.0.0.1:40228
[17:19:58,356] [main] DEBUG (RpcClient.java:1009) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closing ipc connection to localhost/127.0.0.1:40228: Connection refused
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:206)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:529)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:493)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupConnection(RpcClient.java:578)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:868)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.client.MetaScanner.metaScan(MetaScanner.java:144)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.prefetchRegionCache(HConnectionManager.java:1140)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegionInMeta(HConnectionManager.java:1202)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1092)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1049)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.getRegionLocation(HConnectionManager.java:890)
	at org.apache.hadoop.hbase.client.RegionServerCallable.prepare(RegionServerCallable.java:72)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:116)
	at org.apache.hadoop.hbase.client.HTable.get(HTable.java:787)
	at org.apache.hadoop.hbase.examples.customer.GetTable.main(GetTable.java:33)
[17:19:58,358] [main] DEBUG (RpcClient.java:1017) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closed
[17:19:58,358] [main]  INFO (RpcRetryingCaller.java:120) - Call exception, tries=30, retries=35, retryTime=470403ms, msg=row 'Customer,101,99999999999999' on table 'hbase:meta' at region=hbase:meta,,1.1588230740, hostname=localhost,40228,1417408326094, seqNum=0
[17:20:11,700] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:717) - Got ping response for sessionid: 0x14a041ea3cd0030 after 1ms
[17:20:18,533] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 65,4  replyHeader:: 65,262,0  request:: '/hbase/table/hbase:meta,F  response:: #ffffffff000146d61737465723a3436373430ffffffbdffffffa98ffffff83fffffff8ffffffe83fffffffe65042554680,s{36,37,1417408333505,1417408333517,1,0,0,0,31,0,36} 
[17:20:18,537] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 66,4  replyHeader:: 66,262,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a343032323829fffffff153cffffffbf6ffffffd8ffffffc250425546a16a96c6f63616c686f737410ffffffa4ffffffba218ffffffceffffffd3fffffffaffffffa0ffffffa029100,s{39,39,1417408333631,1417408333631,0,0,0,0,61,0,39} 
[17:20:18,537] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[17:20:18,538] [main] DEBUG (RpcClient.java:862) - Connecting to localhost/127.0.0.1:40228
[17:20:18,540] [main] DEBUG (RpcClient.java:1009) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closing ipc connection to localhost/127.0.0.1:40228: Connection refused
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:206)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:529)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:493)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupConnection(RpcClient.java:578)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:868)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.client.MetaScanner.metaScan(MetaScanner.java:144)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.prefetchRegionCache(HConnectionManager.java:1140)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegionInMeta(HConnectionManager.java:1202)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1092)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1049)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.getRegionLocation(HConnectionManager.java:890)
	at org.apache.hadoop.hbase.client.RegionServerCallable.prepare(RegionServerCallable.java:72)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:116)
	at org.apache.hadoop.hbase.client.HTable.get(HTable.java:787)
	at org.apache.hadoop.hbase.examples.customer.GetTable.main(GetTable.java:33)
[17:20:18,542] [main] DEBUG (RpcClient.java:1017) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closed
[17:20:18,542] [main]  INFO (RpcRetryingCaller.java:120) - Call exception, tries=31, retries=35, retryTime=490587ms, msg=row 'Customer,101,99999999999999' on table 'hbase:meta' at region=hbase:meta,,1.1588230740, hostname=localhost,40228,1417408326094, seqNum=0
[17:20:31,884] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:717) - Got ping response for sessionid: 0x14a041ea3cd0030 after 1ms
[17:20:38,713] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 67,4  replyHeader:: 67,262,0  request:: '/hbase/table/hbase:meta,F  response:: #ffffffff000146d61737465723a3436373430ffffffbdffffffa98ffffff83fffffff8ffffffe83fffffffe65042554680,s{36,37,1417408333505,1417408333517,1,0,0,0,31,0,36} 
[17:20:38,718] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 68,4  replyHeader:: 68,262,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a343032323829fffffff153cffffffbf6ffffffd8ffffffc250425546a16a96c6f63616c686f737410ffffffa4ffffffba218ffffffceffffffd3fffffffaffffffa0ffffffa029100,s{39,39,1417408333631,1417408333631,0,0,0,0,61,0,39} 
[17:20:38,720] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[17:20:38,721] [main] DEBUG (RpcClient.java:862) - Connecting to localhost/127.0.0.1:40228
[17:20:38,723] [main] DEBUG (RpcClient.java:1009) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closing ipc connection to localhost/127.0.0.1:40228: Connection refused
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:206)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:529)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:493)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupConnection(RpcClient.java:578)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:868)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.client.MetaScanner.metaScan(MetaScanner.java:144)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.prefetchRegionCache(HConnectionManager.java:1140)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegionInMeta(HConnectionManager.java:1202)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1092)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1049)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.getRegionLocation(HConnectionManager.java:890)
	at org.apache.hadoop.hbase.client.RegionServerCallable.prepare(RegionServerCallable.java:72)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:116)
	at org.apache.hadoop.hbase.client.HTable.get(HTable.java:787)
	at org.apache.hadoop.hbase.examples.customer.GetTable.main(GetTable.java:33)
[17:20:38,726] [main] DEBUG (RpcClient.java:1017) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closed
[17:20:38,726] [main]  INFO (RpcRetryingCaller.java:120) - Call exception, tries=32, retries=35, retryTime=510771ms, msg=row 'Customer,101,99999999999999' on table 'hbase:meta' at region=hbase:meta,,1.1588230740, hostname=localhost,40228,1417408326094, seqNum=0
[17:20:52,062] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:717) - Got ping response for sessionid: 0x14a041ea3cd0030 after 0ms
[17:20:58,764] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 69,4  replyHeader:: 69,262,0  request:: '/hbase/table/hbase:meta,F  response:: #ffffffff000146d61737465723a3436373430ffffffbdffffffa98ffffff83fffffff8ffffffe83fffffffe65042554680,s{36,37,1417408333505,1417408333517,1,0,0,0,31,0,36} 
[17:20:58,768] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 70,4  replyHeader:: 70,262,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a343032323829fffffff153cffffffbf6ffffffd8ffffffc250425546a16a96c6f63616c686f737410ffffffa4ffffffba218ffffffceffffffd3fffffffaffffffa0ffffffa029100,s{39,39,1417408333631,1417408333631,0,0,0,0,61,0,39} 
[17:20:58,770] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[17:20:58,771] [main] DEBUG (RpcClient.java:862) - Connecting to localhost/127.0.0.1:40228
[17:20:58,772] [main] DEBUG (RpcClient.java:1009) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closing ipc connection to localhost/127.0.0.1:40228: Connection refused
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:206)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:529)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:493)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupConnection(RpcClient.java:578)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:868)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.client.MetaScanner.metaScan(MetaScanner.java:144)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.prefetchRegionCache(HConnectionManager.java:1140)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegionInMeta(HConnectionManager.java:1202)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1092)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1049)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.getRegionLocation(HConnectionManager.java:890)
	at org.apache.hadoop.hbase.client.RegionServerCallable.prepare(RegionServerCallable.java:72)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:116)
	at org.apache.hadoop.hbase.client.HTable.get(HTable.java:787)
	at org.apache.hadoop.hbase.examples.customer.GetTable.main(GetTable.java:33)
[17:20:58,795] [main] DEBUG (RpcClient.java:1017) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closed
[17:20:58,795] [main]  INFO (RpcRetryingCaller.java:120) - Call exception, tries=33, retries=35, retryTime=530840ms, msg=row 'Customer,101,99999999999999' on table 'hbase:meta' at region=hbase:meta,,1.1588230740, hostname=localhost,40228,1417408326094, seqNum=0
[17:21:12,109] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:717) - Got ping response for sessionid: 0x14a041ea3cd0030 after 1ms
[17:21:18,843] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 71,4  replyHeader:: 71,262,0  request:: '/hbase/table/hbase:meta,F  response:: #ffffffff000146d61737465723a3436373430ffffffbdffffffa98ffffff83fffffff8ffffffe83fffffffe65042554680,s{36,37,1417408333505,1417408333517,1,0,0,0,31,0,36} 
[17:21:18,846] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 72,4  replyHeader:: 72,262,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a343032323829fffffff153cffffffbf6ffffffd8ffffffc250425546a16a96c6f63616c686f737410ffffffa4ffffffba218ffffffceffffffd3fffffffaffffffa0ffffffa029100,s{39,39,1417408333631,1417408333631,0,0,0,0,61,0,39} 
[17:21:18,847] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[17:21:18,848] [main] DEBUG (RpcClient.java:862) - Connecting to localhost/127.0.0.1:40228
[17:21:18,849] [main] DEBUG (RpcClient.java:1009) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closing ipc connection to localhost/127.0.0.1:40228: Connection refused
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:206)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:529)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:493)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupConnection(RpcClient.java:578)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:868)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.client.MetaScanner.metaScan(MetaScanner.java:144)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.prefetchRegionCache(HConnectionManager.java:1140)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegionInMeta(HConnectionManager.java:1202)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1092)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1049)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.getRegionLocation(HConnectionManager.java:890)
	at org.apache.hadoop.hbase.client.RegionServerCallable.prepare(RegionServerCallable.java:72)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:116)
	at org.apache.hadoop.hbase.client.HTable.get(HTable.java:787)
	at org.apache.hadoop.hbase.examples.customer.GetTable.main(GetTable.java:33)
[17:21:18,851] [main] DEBUG (RpcClient.java:1017) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closed
[17:21:18,852] [main]  INFO (RpcRetryingCaller.java:120) - Call exception, tries=34, retries=35, retryTime=550897ms, msg=row 'Customer,101,99999999999999' on table 'hbase:meta' at region=hbase:meta,,1.1588230740, hostname=localhost,40228,1417408326094, seqNum=0
[17:21:18,860] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[17:21:18,860] [main] DEBUG (RpcClient.java:850) - Not trying to connect to localhost/127.0.0.1:40228 this server is in the failed servers list
[17:21:18,860] [main] DEBUG (RpcClient.java:1009) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closing ipc connection to localhost/127.0.0.1:40228: This server is in the failed servers list: localhost/127.0.0.1:40228
org.apache.hadoop.hbase.ipc.RpcClient$FailedServerException: This server is in the failed servers list: localhost/127.0.0.1:40228
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:853)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1526)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegionInMeta(HConnectionManager.java:1216)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1092)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1049)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.getRegionLocation(HConnectionManager.java:890)
	at org.apache.hadoop.hbase.client.RegionServerCallable.prepare(RegionServerCallable.java:72)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:116)
	at org.apache.hadoop.hbase.client.HTable.get(HTable.java:787)
	at org.apache.hadoop.hbase.examples.customer.GetTable.main(GetTable.java:33)
[17:21:18,861] [main] DEBUG (RpcClient.java:1017) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closed
[17:21:18,861] [main] DEBUG (HConnectionManager.java:1281) - locateRegionInMeta parentTable=hbase:meta, metaLocation={region=hbase:meta,,1.1588230740, hostname=localhost,40228,1417408326094, seqNum=0}, attempt=0 of 35 failed; retrying after sleep of 100 because: This server is in the failed servers list: localhost/127.0.0.1:40228
[17:21:18,862] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 73,4  replyHeader:: 73,262,0  request:: '/hbase/table/hbase:meta,F  response:: #ffffffff000146d61737465723a3436373430ffffffbdffffffa98ffffff83fffffff8ffffffe83fffffffe65042554680,s{36,37,1417408333505,1417408333517,1,0,0,0,31,0,36} 
[17:21:18,865] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 74,4  replyHeader:: 74,262,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a343032323829fffffff153cffffffbf6ffffffd8ffffffc250425546a16a96c6f63616c686f737410ffffffa4ffffffba218ffffffceffffffd3fffffffaffffffa0ffffffa029100,s{39,39,1417408333631,1417408333631,0,0,0,0,61,0,39} 
[17:21:18,970] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 75,4  replyHeader:: 75,262,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a343032323829fffffff153cffffffbf6ffffffd8ffffffc250425546a16a96c6f63616c686f737410ffffffa4ffffffba218ffffffceffffffd3fffffffaffffffa0ffffffa029100,s{39,39,1417408333631,1417408333631,0,0,0,0,61,0,39} 
[17:21:18,974] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 76,4  replyHeader:: 76,262,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a343032323829fffffff153cffffffbf6ffffffd8ffffffc250425546a16a96c6f63616c686f737410ffffffa4ffffffba218ffffffceffffffd3fffffffaffffffa0ffffffa029100,s{39,39,1417408333631,1417408333631,0,0,0,0,61,0,39} 
[17:21:18,975] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[17:21:18,976] [main] DEBUG (RpcClient.java:850) - Not trying to connect to localhost/127.0.0.1:40228 this server is in the failed servers list
[17:21:18,977] [main] DEBUG (RpcClient.java:1009) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closing ipc connection to localhost/127.0.0.1:40228: This server is in the failed servers list: localhost/127.0.0.1:40228
org.apache.hadoop.hbase.ipc.RpcClient$FailedServerException: This server is in the failed servers list: localhost/127.0.0.1:40228
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:853)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.client.MetaScanner.metaScan(MetaScanner.java:144)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.prefetchRegionCache(HConnectionManager.java:1140)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegionInMeta(HConnectionManager.java:1202)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1092)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1049)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.getRegionLocation(HConnectionManager.java:890)
	at org.apache.hadoop.hbase.client.RegionServerCallable.prepare(RegionServerCallable.java:72)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:116)
	at org.apache.hadoop.hbase.client.HTable.get(HTable.java:787)
	at org.apache.hadoop.hbase.examples.customer.GetTable.main(GetTable.java:33)
[17:21:18,978] [main] DEBUG (RpcClient.java:1017) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closed
[17:21:19,283] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 77,4  replyHeader:: 77,262,0  request:: '/hbase/table/hbase:meta,F  response:: #ffffffff000146d61737465723a3436373430ffffffbdffffffa98ffffff83fffffff8ffffffe83fffffffe65042554680,s{36,37,1417408333505,1417408333517,1,0,0,0,31,0,36} 
[17:21:19,287] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 78,4  replyHeader:: 78,262,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a343032323829fffffff153cffffffbf6ffffffd8ffffffc250425546a16a96c6f63616c686f737410ffffffa4ffffffba218ffffffceffffffd3fffffffaffffffa0ffffffa029100,s{39,39,1417408333631,1417408333631,0,0,0,0,61,0,39} 
[17:21:19,288] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[17:21:19,289] [main] DEBUG (RpcClient.java:850) - Not trying to connect to localhost/127.0.0.1:40228 this server is in the failed servers list
[17:21:19,290] [main] DEBUG (RpcClient.java:1009) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closing ipc connection to localhost/127.0.0.1:40228: This server is in the failed servers list: localhost/127.0.0.1:40228
org.apache.hadoop.hbase.ipc.RpcClient$FailedServerException: This server is in the failed servers list: localhost/127.0.0.1:40228
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:853)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.client.MetaScanner.metaScan(MetaScanner.java:144)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.prefetchRegionCache(HConnectionManager.java:1140)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegionInMeta(HConnectionManager.java:1202)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1092)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1049)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.getRegionLocation(HConnectionManager.java:890)
	at org.apache.hadoop.hbase.client.RegionServerCallable.prepare(RegionServerCallable.java:72)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:116)
	at org.apache.hadoop.hbase.client.HTable.get(HTable.java:787)
	at org.apache.hadoop.hbase.examples.customer.GetTable.main(GetTable.java:33)
[17:21:19,293] [main] DEBUG (RpcClient.java:1017) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closed
[17:21:19,800] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 79,4  replyHeader:: 79,262,0  request:: '/hbase/table/hbase:meta,F  response:: #ffffffff000146d61737465723a3436373430ffffffbdffffffa98ffffff83fffffff8ffffffe83fffffffe65042554680,s{36,37,1417408333505,1417408333517,1,0,0,0,31,0,36} 
[17:21:19,803] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 80,4  replyHeader:: 80,262,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a343032323829fffffff153cffffffbf6ffffffd8ffffffc250425546a16a96c6f63616c686f737410ffffffa4ffffffba218ffffffceffffffd3fffffffaffffffa0ffffffa029100,s{39,39,1417408333631,1417408333631,0,0,0,0,61,0,39} 
[17:21:19,804] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[17:21:19,805] [main] DEBUG (RpcClient.java:850) - Not trying to connect to localhost/127.0.0.1:40228 this server is in the failed servers list
[17:21:19,805] [main] DEBUG (RpcClient.java:1009) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closing ipc connection to localhost/127.0.0.1:40228: This server is in the failed servers list: localhost/127.0.0.1:40228
org.apache.hadoop.hbase.ipc.RpcClient$FailedServerException: This server is in the failed servers list: localhost/127.0.0.1:40228
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:853)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.client.MetaScanner.metaScan(MetaScanner.java:144)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.prefetchRegionCache(HConnectionManager.java:1140)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegionInMeta(HConnectionManager.java:1202)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1092)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1049)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.getRegionLocation(HConnectionManager.java:890)
	at org.apache.hadoop.hbase.client.RegionServerCallable.prepare(RegionServerCallable.java:72)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:116)
	at org.apache.hadoop.hbase.client.HTable.get(HTable.java:787)
	at org.apache.hadoop.hbase.examples.customer.GetTable.main(GetTable.java:33)
[17:21:19,816] [main] DEBUG (RpcClient.java:1017) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closed
[17:21:20,821] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 81,4  replyHeader:: 81,262,0  request:: '/hbase/table/hbase:meta,F  response:: #ffffffff000146d61737465723a3436373430ffffffbdffffffa98ffffff83fffffff8ffffffe83fffffffe65042554680,s{36,37,1417408333505,1417408333517,1,0,0,0,31,0,36} 
[17:21:20,825] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 82,4  replyHeader:: 82,262,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a343032323829fffffff153cffffffbf6ffffffd8ffffffc250425546a16a96c6f63616c686f737410ffffffa4ffffffba218ffffffceffffffd3fffffffaffffffa0ffffffa029100,s{39,39,1417408333631,1417408333631,0,0,0,0,61,0,39} 
[17:21:20,825] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[17:21:20,826] [main] DEBUG (RpcClient.java:850) - Not trying to connect to localhost/127.0.0.1:40228 this server is in the failed servers list
[17:21:20,827] [main] DEBUG (RpcClient.java:1009) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closing ipc connection to localhost/127.0.0.1:40228: This server is in the failed servers list: localhost/127.0.0.1:40228
org.apache.hadoop.hbase.ipc.RpcClient$FailedServerException: This server is in the failed servers list: localhost/127.0.0.1:40228
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:853)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.client.MetaScanner.metaScan(MetaScanner.java:144)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.prefetchRegionCache(HConnectionManager.java:1140)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegionInMeta(HConnectionManager.java:1202)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1092)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1049)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.getRegionLocation(HConnectionManager.java:890)
	at org.apache.hadoop.hbase.client.RegionServerCallable.prepare(RegionServerCallable.java:72)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:116)
	at org.apache.hadoop.hbase.client.HTable.get(HTable.java:787)
	at org.apache.hadoop.hbase.examples.customer.GetTable.main(GetTable.java:33)
[17:21:20,829] [main] DEBUG (RpcClient.java:1017) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closed
[17:21:22,842] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 83,4  replyHeader:: 83,262,0  request:: '/hbase/table/hbase:meta,F  response:: #ffffffff000146d61737465723a3436373430ffffffbdffffffa98ffffff83fffffff8ffffffe83fffffffe65042554680,s{36,37,1417408333505,1417408333517,1,0,0,0,31,0,36} 
[17:21:22,845] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 84,4  replyHeader:: 84,262,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a343032323829fffffff153cffffffbf6ffffffd8ffffffc250425546a16a96c6f63616c686f737410ffffffa4ffffffba218ffffffceffffffd3fffffffaffffffa0ffffffa029100,s{39,39,1417408333631,1417408333631,0,0,0,0,61,0,39} 
[17:21:22,846] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[17:21:22,847] [main] DEBUG (RpcClient.java:862) - Connecting to localhost/127.0.0.1:40228
[17:21:22,848] [main] DEBUG (RpcClient.java:1009) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closing ipc connection to localhost/127.0.0.1:40228: Connection refused
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:206)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:529)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:493)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupConnection(RpcClient.java:578)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:868)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.client.MetaScanner.metaScan(MetaScanner.java:144)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.prefetchRegionCache(HConnectionManager.java:1140)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegionInMeta(HConnectionManager.java:1202)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1092)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1049)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.getRegionLocation(HConnectionManager.java:890)
	at org.apache.hadoop.hbase.client.RegionServerCallable.prepare(RegionServerCallable.java:72)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:116)
	at org.apache.hadoop.hbase.client.HTable.get(HTable.java:787)
	at org.apache.hadoop.hbase.examples.customer.GetTable.main(GetTable.java:33)
[17:21:22,850] [main] DEBUG (RpcClient.java:1017) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closed
[17:21:26,882] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 85,4  replyHeader:: 85,262,0  request:: '/hbase/table/hbase:meta,F  response:: #ffffffff000146d61737465723a3436373430ffffffbdffffffa98ffffff83fffffff8ffffffe83fffffffe65042554680,s{36,37,1417408333505,1417408333517,1,0,0,0,31,0,36} 
[17:21:26,886] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 86,4  replyHeader:: 86,262,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a343032323829fffffff153cffffffbf6ffffffd8ffffffc250425546a16a96c6f63616c686f737410ffffffa4ffffffba218ffffffceffffffd3fffffffaffffffa0ffffffa029100,s{39,39,1417408333631,1417408333631,0,0,0,0,61,0,39} 
[17:21:26,887] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[17:21:26,888] [main] DEBUG (RpcClient.java:862) - Connecting to localhost/127.0.0.1:40228
[17:21:26,889] [main] DEBUG (RpcClient.java:1009) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closing ipc connection to localhost/127.0.0.1:40228: Connection refused
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:206)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:529)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:493)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupConnection(RpcClient.java:578)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:868)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.client.MetaScanner.metaScan(MetaScanner.java:144)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.prefetchRegionCache(HConnectionManager.java:1140)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegionInMeta(HConnectionManager.java:1202)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1092)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1049)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.getRegionLocation(HConnectionManager.java:890)
	at org.apache.hadoop.hbase.client.RegionServerCallable.prepare(RegionServerCallable.java:72)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:116)
	at org.apache.hadoop.hbase.client.HTable.get(HTable.java:787)
	at org.apache.hadoop.hbase.examples.customer.GetTable.main(GetTable.java:33)
[17:21:26,891] [main] DEBUG (RpcClient.java:1017) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closed
[17:21:36,978] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 87,4  replyHeader:: 87,262,0  request:: '/hbase/table/hbase:meta,F  response:: #ffffffff000146d61737465723a3436373430ffffffbdffffffa98ffffff83fffffff8ffffffe83fffffffe65042554680,s{36,37,1417408333505,1417408333517,1,0,0,0,31,0,36} 
[17:21:36,979] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:717) - Got ping response for sessionid: 0x14a041ea3cd0030 after 4ms
[17:21:36,982] [main-SendThread(localhost:2181)] DEBUG (ClientCnxn.java:818) - Reading reply sessionid:0x14a041ea3cd0030, packet:: clientPath:null serverPath:null finished:false header:: 88,4  replyHeader:: 88,262,0  request:: '/hbase/meta-region-server,F  response:: #ffffffff0001a726567696f6e7365727665723a343032323829fffffff153cffffffbf6ffffffd8ffffffc250425546a16a96c6f63616c686f737410ffffffa4ffffffba218ffffffceffffffd3fffffffaffffffa0ffffffa029100,s{39,39,1417408333631,1417408333631,0,0,0,0,61,0,39} 
[17:21:36,983] [main] DEBUG (RpcClient.java:428) - Use SIMPLE authentication for service ClientService, sasl=false
[17:21:36,983] [main] DEBUG (RpcClient.java:862) - Connecting to localhost/127.0.0.1:40228
[17:21:36,985] [main] DEBUG (RpcClient.java:1009) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closing ipc connection to localhost/127.0.0.1:40228: Connection refused
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:206)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:529)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:493)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupConnection(RpcClient.java:578)
	at org.apache.hadoop.hbase.ipc.RpcClient$Connection.setupIOstreams(RpcClient.java:868)
	at org.apache.hadoop.hbase.ipc.RpcClient.getConnection(RpcClient.java:1543)
	at org.apache.hadoop.hbase.ipc.RpcClient.call(RpcClient.java:1442)
	at org.apache.hadoop.hbase.ipc.RpcClient.callBlockingMethod(RpcClient.java:1661)
	at org.apache.hadoop.hbase.ipc.RpcClient$BlockingRpcChannelImplementation.callBlockingMethod(RpcClient.java:1719)
	at org.apache.hadoop.hbase.protobuf.generated.ClientProtos$ClientService$BlockingStub.get(ClientProtos.java:30363)
	at org.apache.hadoop.hbase.protobuf.ProtobufUtil.getRowOrBefore(ProtobufUtil.java:1546)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:717)
	at org.apache.hadoop.hbase.client.HTable$2.call(HTable.java:715)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:117)
	at org.apache.hadoop.hbase.client.HTable.getRowOrBefore(HTable.java:721)
	at org.apache.hadoop.hbase.client.MetaScanner.metaScan(MetaScanner.java:144)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.prefetchRegionCache(HConnectionManager.java:1140)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegionInMeta(HConnectionManager.java:1202)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1092)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.locateRegion(HConnectionManager.java:1049)
	at org.apache.hadoop.hbase.client.HConnectionManager$HConnectionImplementation.getRegionLocation(HConnectionManager.java:890)
	at org.apache.hadoop.hbase.client.RegionServerCallable.prepare(RegionServerCallable.java:72)
	at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:116)
	at org.apache.hadoop.hbase.client.HTable.get(HTable.java:787)
	at org.apache.hadoop.hbase.examples.customer.GetTable.main(GetTable.java:33)
[17:21:36,986] [main] DEBUG (RpcClient.java:1017) - IPC Client (553434543) connection to localhost/127.0.0.1:40228 from kylin: closed

